include/seqan3/alphabet/nucleotide/dna15.hpp

```diff
@@ -152,7 +152,7 @@ struct dna15
     //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept and have the same \link value_size \endlink.
     template <typename other_nucl_type>
     //!\cond
-        requires nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
+        requires !std::Same<dna15, other_nucl_type> && nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
```

Comment:
maybe put the `nucleotide_concept` into the template declaration (short concept notation) so that this line becomes shorter? 
Hm I see that you explicitly changed the short notation to the long one below.. why ?

Reply:
The order is important, because we have an infinite recursion (on clang) if we first go into `nucleotide_concept` and afterwards check the `Same` condition.

Reply:
Is this a bug in clang?

Reply:
or is this "The correct way of doing it"?

Reply:
Sry for answering late, this apparently is the correct way in the current standard.